### DISCUSSION TITLE
2.5.5 with 2.5.6  boards.txt, platform.txt - bad python3 location ??

### DISCUSSION THREAD

**Discussion by apws** [SpenceKonde/megaTinyCore#609] [Q&A] [answered]
hi, although, 2.5.6 is not yet propagated to arduino, I peek into arduino15 folder for quick fix and replaced boards.txt and platform.txt with the new ones from github and I can confirm that now the extra params for optiboot avrdude flash is parsed OK.

BUT Another issue I had was burning bootlader with correctly installed 2.5.5 because python3 cant be found at location:

C:\Users\petra\AppData\Local\Arduino15\packages\megaTinyCore\hardware\megaavr\2.5.5/tools/python3/python3 -u C:\Users\petra\AppData\Local\Arduino15\packages\megaTinyCore\hardware\megaavr\2.5.5/tools/prog.py -t uart -u COM10 -b 230400 -d attiny3226 --fuses 0:0x00 1:0x00 2:0x02 5:0b11111110 6:0x04 7:0x00 8:0x02 -fC:\Users\petra\AppData\Local\Arduino15\packages\megaTinyCore\hardware\megaavr\2.5.5/bootloaders/hex/optiboot_tx2z_rst.hex -a write 

as here is NOT python3:
Arduino15\packages\megaTinyCore\hardware\megaavr\2.5.5/tools/python3/python3

it is placed here:
Arduino15\packages\megaTinyCore\tools\python3\3.7.2-post1\python3.exe

so, out of megaTinyCore\hardware\....

in megaTinyCore platform.txt is this:
tools.serialupdi.cmd={runtime.platform.path}/tools/python3/python3
#REMOVE#tools.serialupdi.cmd={runtime.tools.python3.path}/python3

BEWARE THAT in DxCore platform.txt is this: (and there is NO ANY python3 in tools)
#tools.serialupdi.cmd={runtime.platform.path}/tools/python3/python3
tools.serialupdi.cmd={runtime.tools.python3.path}/python3

2.5.5 location of python3.exe is this, so even in some versioned subfolder is THIS the {runtime.tools.python3.path} ???
so rather in megaTinyAVR also fix platform.txt to this as in DxCore? will try
megaTinyCore\tools\python3\3.7.2-post1

YA... correct for megaTinyAVR is this in platform.txt too ......
tools.serialupdi.cmd={runtime.tools.python3.path}/python3

**Comment by apws** [comment]
BUT now its already in arduino updates and it works well after update :-) OKAY

BUT !! ... it seems my fix caused this as in 2.5.6 platform.txt is still this:
tools.serialupdi.cmd={runtime.platform.path}/tools/python3/python3
#REMOVE#tools.serialupdi.cmd={runtime.tools.python3.path}/python3

I am LOST too ...

**Comment by SpenceKonde** [comment, marked as answer]
The github version of the core is meant to run from (sketchbook)/hardware/megaTinyCore, with manual intallation of python in (sketchbook)/hardware/megaTinyCore/megaavr/tools/python3  as described  in https://github.com/SpenceKonde/megaTinyCore/blob/master/megaavr/tools/ManualPython.md 
When the packages for board manager installation are built, the release script searches platform.txt and  ~deletes~ comments out any line starting with tools.serialupdi.cmd,  Then in searches for #REMOVE# and removes every instance of that string. 

(the same script also deletes extras, which cuts the size of megaTinyCore and DxCore's binaries from 10mb and 40mb to around 700k each. Nobody is going to dig up the documentation inside the installed board manager package itself when they could view it on github, but the pinout diagrams make up the lions share of the repo size)

https://github.com/SpenceKonde/ReleaseScripts/blob/master/build_megaTinyCore.sh

It was originally imagined that the core would have a static description, except that when I release 6 consecutive bad version of the core within under a month, I need to tell people which one to use.
